src/WooCommerce.php

```diff
@@ -561,6 +561,24 @@ public static function woocommerce_display_item_meta($html, $item, $args) {
     return $html;
   }
 
+  /**
+   * Removes SKU from order item name, added by woocommerce-german-market.
+   *
+   * @implements wgm_email_after_item_name
+   */
+  public static function wgm_email_after_item_name($order_item) {
+    ob_start();
+  }
+
+  /**
+   * Removes SKU from order item name, added by woocommerce-german-market.
+   *
+   * @implements woocommerce_order_item_meta_start
+   */
+  public static function woocommerce_order_item_meta_start($item_id, $item, $order, $plain_text) {
+    echo preg_replace('@\s*\(#.+\)@', '', ob_get_clean());
```

Comment:
If installed on a completely different setup, this could match false positives. Doesn't WGM also add this in a filter/action hook that can be unregistered as well?

Reply:
> If installed on a completely different setup, this could match false positives.
I am checking for WGM to be present before setting up the filter chain. See https://github.com/netzstrategen/wordpress-shop-standards/blob/f2cf3d2d07d9e1786e4e8606eef69b4ae4a516ca/src/Plugin.php#L111. Do you think this might be not safe enough?

> Doesn't WGM also add this in a filter/action hook that can be unregistered as well?
Not that I have seen. Adding SKU to item name seems to be controlled by a WGM backend setting, but:
1. I was not able to find a UI control for that setting in the backend plugin settings
2. The setting seems not to be read before using it in the "New Order" email template.
2. This setting effects are not limited to the case we are trying to fix. Changing it would modify other results. I.e. PDF refund documents.

Reply:
The templates seem to conditionally output the SKU based on `show_sku`:
https://github.com/netzstrategen/gaco/blob/df10d723c57ed566b4038b528b507b6616ffff6d/wp-content/plugins/woocommerce/templates/emails/email-order-items.php#L53-L55

`show_sku` is passed from:
https://github.com/netzstrategen/gaco/blob/858888251d7f559c19be82b5bbf4a0cdd7548279/wp-content/plugins/woocommerce-german-market/templates/woocommerce-german-market/emails/plain/customer-confirm-order.php#L40
https://github.com/netzstrategen/gaco/blob/858888251d7f559c19be82b5bbf4a0cdd7548279/wp-content/plugins/woocommerce-german-market/templates/woocommerce-german-market/emails/customer-confirm-order.php#L51

This invokes the filter hook `woocommerce_email_order_items_args` in
https://github.com/netzstrategen/gaco/blob/858888251d7f559c19be82b5bbf4a0cdd7548279/wp-content/plugins/woocommerce/includes/wc-template-functions.php#L2993

Isn't it sufficient to set that parameter to false under the desired conditions/scenarios?

Reply:
@sun Thanks a lot, I missed that. I will fix the approach considering your suggestions.